scilpy/utils/scilpy_bot.py

```diff
@@ -0,0 +1,345 @@
+    dot or newline. If there is no dot or newline, return the full string as
+    the first sentence, and None as the remaining text.
+
+    Parameters
+    ----------
+    text : str
+        Text to parse.
+
+    Returns
+    -------
+    first_sentence : str
+        The first sentence, or the full text if no dot or newline was found.
+    remaining : str
+        Everything after the first sentence.
+
+    """
+    candidates = ['. ', '.\n']
+    sentence_idx = -1
+    for candidate in candidates:
+        idx = text.find(candidate)
+        if idx != -1 and idx < sentence_idx or sentence_idx == -1:
+            sentence_idx = idx
+
+    split_idx = (sentence_idx + 1) or None
+    sentence = text[:split_idx]
+    remaining = text[split_idx:] if split_idx else ""
+    return sentence, remaining
+
+
+def _stem_keywords(keywords):
+    """
+    Stem a list of keywords using PorterStemmer.
+
+    Parameters
+    ----------
+    keywords : list of str
+        Keywords to be stemmed.
+
+    Returns
+    -------
+    list of str
+        Stemmed keywords.
+    """
+    return [stemmer.stem(keyword) for keyword in keywords]
+
+
+def _stem_text(text):
+    """
+    Stem all words in a text using PorterStemmer.
+
+    Parameters
+    ----------
+    text : str
+        Text to be stemmed.
+
+    Returns
+    -------
+    str
+        Stemmed text.
+    """
+    words = nltk.word_tokenize(text)
+    return ' '.join([stemmer.stem(word) for word in words])
+
+
+def _stem_phrase(phrase):
+    """
+    Stem all words in a phrase using PorterStemmer.
+
+    Parameters
+    ----------
+    phrase : str
+        Phrase to be stemmed.
+
+    Returns
+    -------
+    str
+        Stemmed phrase.
+    """
+    words = phrase.split()
+    return ' '.join([stemmer.stem(word) for word in words])
+
+
+def _generate_help_files():
+    """
+    This function iterates over all Python scripts in the 'scripts' directory,
+    runs each script with the '--h' flag to generate help text,
+    and saves the output in the '.hidden' directory.
+
+    By doing this, we can precompute the help outputs for each script,
+    which can be useful for faster searches.
+
+    If a help file already exists for a script, the script is skipped,
+    and the existing help file is left unchanged.
+
+    The help output is saved in a hidden directory to avoid clutter in
+    the main scripts directory.
+    """
+
+    scripts_dir = pathlib.Path(__file__).parent.parent.parent / 'scripts'
+
+    scripts = [script for script in scripts_dir.glob('*.py')
+               if script.name not in ['__init__.py',
+                                      'scil_search_keywords.py']]
+    total_scripts = len(scripts)
+
+    # Hidden directory to store help files
+    hidden_dir = scripts_dir / '.hidden'
+    hidden_dir.mkdir(exist_ok=True)
+
+    # Iterate over all scripts and generate help files
+    for idx, script in enumerate(tqdm(scripts, desc="Generating help files", total=total_scripts), start=1):
```

Comment:
I think the enumerate should be inside the TQDM

Also, maybe it is because of the print below, but right now the progress bar is being re-printed every iteration, going against the idea of using TQDM

![image](https://github.com/user-attachments/assets/d9c31926-c290-4a8b-a0ae-52aee63d7d8a)

It should only show a progress bar on one line, not taking multiple lines.